src/pages/ValidateLoginPage/index.js

```diff
@@ -0,0 +1,65 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import {
+    propTypes as validateLinkPropTypes,
+    defaultProps as validateLinkDefaultProps,
+} from './validateLinkPropTypes';
+import * as User from '../../libs/actions/User';
+import FullScreenLoadingIndicator from '../../components/FullscreenLoadingIndicator';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as Session from '../../libs/actions/Session';
+import Permissions from '../../libs/Permissions';
+import Navigation from '../../libs/Navigation/Navigation';
+
+const propTypes = {
+    /** The accountID and validateCode are passed via the URL */
+    route: validateLinkPropTypes,
+
+    /** List of betas available to current user */
+    betas: PropTypes.arrayOf(PropTypes.string),
+};
+
+const defaultProps = {
+    route: validateLinkDefaultProps,
+    betas: [],
+};
+
+class ValidateLoginPage extends Component {
+    componentDidMount() {
+        if (Permissions.canUsePasswordlessLogins(this.props.betas)) {
+            if (lodashGet(this.props, 'session.authToken', null)) {
+                // If already signed in, do not show the validate code if not on web,
+                // because we don't want to block the user with the interstitial page.
+                Navigation.goBack(false);
+            } else {
+                Session.signInWithValidateCodeAndNavigate(this.accountID(), this.validateCode());
+            }
+        } else {
+            User.validateLogin(this.accountID(), this.validateCode());
+        }
+    }
+
+    /**
+     * @returns {String}
+     */
+    accountID = () => lodashGet(this.props.route.params, 'accountID', '');
```

Comment:
Typically, it is recommended to prefix functions that retrieve information with "get", so perhaps we could consider renaming `accountID()` and `validateCode()` to `getAccountID()` and `getValidateCode()` respectively?

Additionally, I could be out of the loop on our style guide, but I think we have not yet adopted instance class properties. 

I understand it is less verbose than having to declare things in the constructor and people are often confused about `this` in React class components - but can we update to be more consistent with the current styles?

Reply:
Hey @marcaaron ! Thanks for guiding me on following the style! I am working on a follow up PR which touches this class. It should be ready for review. I am adding your suggestions in the [new PR](https://github.com/Expensify/App/pull/15505).